FWCore/Catalog/interface/FileLocator.h

```diff
@@ -1,22 +1,27 @@
 #ifndef FWCore_Catalog_FileLocator_h
 #define FWCore_Catalog_FileLocator_h
 
+#include "FWCore/Catalog/interface/SiteLocalConfig.h"
 #include <string>
 #include <list>
 #include <map>
 #include <utility>
 #include <regex>
 #include "tinyxml2.h"
+#include <boost/property_tree/ptree.hpp>
 
 namespace edm {
 
   class FileLocator {
   public:
+    explicit FileLocator(edm::CatalogAttributes const& catAttr, unsigned iCatalog = 0);
     explicit FileLocator(std::string const& catUrl, unsigned iCatalog = 0);
+
     ~FileLocator();
 
-    std::string pfn(std::string const& ilfn) const;
-    std::string lfn(std::string const& ipfn) const;
+    std::string pfn(std::string const& ilfn, edm::CatalogType catType = edm::CatalogType::TrivialCatalog) const;
```

Comment:
I would consider making the `catType` a mandatory parameter, i.e. drop the default value. Not a big deal though if the `TrivialCatalog` gets cleaned up soonish afterwards. What caught my eye was the default value being `TrivialCatalog` while in the bigger picture the `RucioCatalog` is being made the default.

Reply:
Yes, I agree. It is better to mandate ```catType```. (Initially, in attempt to keep original codes which use TrivialCatalog unchanged, I put the default  choice of TrivalCatalog here, but now it is inappropriate since we are going to chose RucioCatalog as default soon).
Is ```FWCore/Catalog/test/FileLocator_t.cpp``` used somewhere? If it is uses for a test somewhere, we might need to modify it to check the file locator using RucioCatalog. Currently, it only check ones using TrivialCatalog, for example:
https://github.com/cms-sw/cmssw/blob/693d0c951b023a1c2ba1daac9832f4985f81ccdd/FWCore/Catalog/test/FileLocator_t.cpp#L79
Should I address this in this PR as well?

Reply:
@makortel  I have written tests for RucioCatalog in ```FWCore/Catalog/test/FileLocator_t.cpp```. Do you know how to run this code locally to check if the tests work as expected?

Reply:
There are several options
* run `${CMSSW_BASE}/test/${SCRAM_ARCH}/edmCatalogFileLocator_t` for just this test
* `cd ${CMSSW_BASE}/src/FWCore/Catalog; scram b runtests` to run all tests in `FWCore/Catalog` (well, it has only this test)
* In `${CMSSW_BASE}/src` run `scram b runtests` to run all tests in all checked out packages